replica/database.hh

```diff
@@ -402,7 +402,7 @@ private:
     locator::effective_replication_map_ptr _erm;
     lw_shared_ptr<const storage_options> _storage_opts;
     mutable table_stats _stats;
-    mutable db::view::stats _view_stats;
+    mutable std::unique_ptr<db::view::stats> _view_stats;
```

Comment:
Unless I'm missing something, this looks wrong to me. view_stats is **not** the metrics registration, but rather the counters themselves. We have to keep those alive all the time, we turn this pointer to null, and don't need to...
I suspect that if you ran a test that has materialized views, you would see a crash if materialized view statistics are updated in parallel with deleting a table.

Reply:
I think we're good with what Avi proposed in his comment - https://github.com/scylladb/scylladb/pull/13971#discussion_r1199768423
I already made the adjustments. BTW. I run `test_materialized_view.py` with the `metrics_group::clear()` fix and all of them passed, though I'm not sure if the "bug" you mentioned would occur just by running an unmodified version of these tests.